### PR TITLE
Add Support for Bahrain me-south-1 AWS region

### PR DESCRIPTION
**Pull request by patrickdillon**
Updates validation to allow installing in the me-south-1 Bahrain region in AWS.

/hold
Depends on RHCOS images being published for this region. 

JIRA: [CORS-1313](https://issues.redhat.com/browse/CORS-1313)

cc @miabbott

### CODE REVIEW

**Comment by patrickdillon** [comment]
support for the me-south-1 region was added to the AWS Terraform provider in v2.22 

This PR needs #2676

**Comment by patrickdillon** [comment]
Gets to 100% but image-registry fails: 
```
DEBUG Still waiting for the cluster to initialize: Working towards 4.4.0-0.nightly-2019-12-20-210709: 100% complete, waiting on image-registry 
DEBUG Still waiting for the cluster to initialize: Cluster operator image-registry is still updating 
INFO Cluster operator image-registry Available is False with NoReplicasAvailable: The deployment does not have available replicas 
INFO Cluster operator image-registry Progressing is True with DeploymentNotCompleted: The deployment has not completed 
INFO Cluster operator insights Disabled is False with :  
```

```

**Comment by patrickdillon** [comment]
/hold cancel
I think issue in https://github.com/openshift/installer/pull/2826#issuecomment-57027367 is unrelated to this PR, but does need to be resolved.

**Comment by patrickdillon** [comment]
/hold
Seems like this will indeed depend on image registry adding me-south-1 region:
```
$ oc get pods -n openshift-image-registry
NAME                                               READY   STATUS             RESTARTS   AGE
cluster-image-registry-operator-7796d787ff-clggk   2/2     Running            0          68m
image-registry-79c5b59b8-jqqct                     0/1     CrashLoopBackOff   17         67m
image-registry-dcfb7bb6f-kcw2b                     0/1     CrashLoopBackOff   17         67m
node-ca-2s9lb                                      1/1     Running            0          65m
node-ca-4vq26                                      1/1     Running            0          64m
node-ca-67kfz                                      1/1     Running            0          67m
node-ca-j4985                                      1/1     Running            0          67m
node-ca-s2v8g                                      1/1     Running            0          65m
node-ca-xr8tx                                      1/1     Running            0          67m
$ oc logs image-registry-79c5b59b8-jqqct -n openshift-image-registry
time="2020-01-03T14:59:40.84698399Z" level=info msg="start registry" distribution_version=v2.6.0+unknown go.version=go1.12.12 openshift_version=v4.4.0-201912200058+2c3bfc0-dirty
time="2020-01-03T14:59:40.847679595Z" level=info msg="caching project quota objects with TTL 1m0s" go.version=go1.12.12
panic: Invalid region provided: me-south-1

goroutine 1 [running]:
github.com/openshift/image-registry/vendor/github.com/docker/distribution/registry/handlers.NewApp(0x1e30260, 0xc000048090, 0xc000291c00, 0xc00009d9e0)
	/go/src/github.com/openshift/image-registry/vendor/github.com/docker/distribution/registry/handlers/app.go:127 +0x31ac
github.com/openshift/image-registry/pkg/dockerregistry/server/supermiddleware.NewApp(0x1e30260, 0xc000048090, 0xc000291c00, 0x1e37de0, 0xc0002d9950, 0x1e42c00)
	/go/src/github.com/openshift/image-registry/pkg/dockerregistry/server/supermiddleware/app.go:96 +0x85
github.com/openshift/image-registry/pkg/dockerregistry/server.NewApp(0x1e30260, 0xc000048090, 0x1e03aa0, 0xc00010c2e0, 0xc000291c00, 0xc000206280, 0x0, 0x0, 0x0, 0xc00004e000)
	/go/src/github.com/openshift/image-registry/pkg/dockerregistry/server/app.go:138 +0x2d4
github.com/openshift/image-registry/pkg/cmd/dockerregistry.NewServer(0x1e30260, 0xc000048090, 0xc000291c00, 0xc000206280, 0x0, 0x0, 0x1e6fd00)
	/go/src/github.com/openshift/image-registry/pkg/cmd/dockerregistry/dockerregistry.go:210 +0x1c2
github.com/openshift/image-registry/pkg/cmd/dockerregistry.Execute(0x1deda00, 0xc00010c020)
	/go/src/github.com/openshift/image-registry/pkg/cmd/dockerregistry/dockerregistry.go:164 +0xa42
main.main()
	/go/src/github.com/openshift/image-registry/cmd/dockerregistry/main.go:93 +0x49c
```

**Comment by patrickdillon** [comment]
/hold cancel
Despite [image-registry issue](https://bugzilla.redhat.com/show_bug.cgi?id=1787604) installer code should be ready for review/merge

**Comment by wking** [comment]
> support for the me-south-1 region was added to the AWS Terraform provider in v2.22
>
> This PR needs #2676

And that PR is still open, and in turn [depends on Go 1.13][1].  4.4+ CI builds use 1.13 since openshift/release#5928, and 4.4+ ART builds use 1.13 since openshift/ocp-build-data@a40715256052f3efa8e15040a3b7317ada60a043, so we should be clear to move forward with the Terraform bump and then circle back to this one.

[1]: https://github.com/openshift/installer/pull/2676#issuecomment-554230975

**Comment by patrickdillon** [comment]
> And that PR is still open, and in turn depends on Go 1.13. 4.4+ CI builds use 1.13 since openshift/release#5928, and 4.4+ ART builds use 1.13 since openshift/ocp-build-data@a407152, so we should be clear to move forward with the Terraform bump and then circle back to this one.

@wking sorry I addressed this on slack but forgot to update the PR. 82f1ee9520b0fd8f3abac8d9ed8cce5fc8ad4f87 was my temporary solution to this in 4.4. I think it will be a permanent solution in 4.3 & 4.2

File: Gopkg.toml [hunk 1]
```diff
@@ -136,3 +136,7 @@ required = [
   branch = "master"
   name = "github.com/metal3-io/baremetal-operator"
   source = "https://github.com/openshift/baremetal-operator"
+
+[[override]]
+  name = "gopkg.in/fsnotify.v1"
+  source = "https://github.com/fsnotify/fsnotify"
```
Comment (wking) [inline Gopkg.toml]: Why do you need this override?

Comment (patrickdillon) [reply Gopkg.toml]: Removed this. Was added by mistake.

**Pull request merged**